classes/event_processor.php

```diff
@@ -216,6 +216,13 @@ private function process_realtime_workflow($workflow, $evententry) {
             foreach ($steps as $step) {
                 try {
                     $outertransaction = $DB->is_transaction_started();
+
+                    // Special case, the debounce step should never process realtime.
+                    // Bail here, and move to the cron queue.
+                    if ($step->stepclass === '\tool_trigger\steps\debounce\debounce_step') {
```

Comment:
could this be inside the execute function of the step?

Reply:
Yeah, the identification can be based on the cron user, so we know the env

Reply:
Or we could do the even smarter thing and use CLI_SCRIPT -_- . Fixed :)